### PR TITLE
annealing kl and temperature losses

### PR DESCRIPTION
**Pull request by blandry**

### CODE REVIEW

File: mie2c/e2c.py [hunk 1]
```diff
@@ -135,6 +140,7 @@ def forward(self, x):
             # x = self.batch_norm_layers[ii](x)
             if self.conv_activation:
                 x = self.conv_activation(x)
+        x = self.sigmoid(x)
```
Comment (acauligi) [inline mie2c/e2c.py:143]: Should the _sigmoid_ activation only be applied for the loss function so that the forward pass can still yield an RGB image?

File: mie2c/e2c.py [hunk 2]
```diff
@@ -356,23 +362,102 @@ def predict(self, X, U):
...
-    x_reconst_loss = F.binary_cross_entropy(nn.Sigmoid()(x_dec), nn.Sigmoid()(x), reduction='sum')
-    x_next_reconst_loss = F.binary_cross_entropy(nn.Sigmoid()(x_next_pred_dec), nn.Sigmoid()(x_next), reduction='sum')
+                 Qz, Qz_next, Qz_next_pred,
+                 use_l2=False):
+    if use_l2:
+        x_reconst_loss = (x_dec - x).pow(2).sum(dim=[1,2,3])
+        x_next_reconst_loss = (x_next_pred_dec - x_next).pow(2).sum(dim=[1,2,3])
+    else:
+        bce_loss = nn.BCELoss(reduction='mean')
+        x_reconst_loss = bce_loss(torch.clamp(x_dec, 0., 1.), x)
```
Comment (acauligi) [inline mie2c/e2c.py:375]: Shouldn't _clamp_ be applied to x and x_next to ensure the image data is normalized?

File: mie2c/e2c.py (hunk 1 above)
Comment (blandry) [reply mie2c/e2c.py:143]: you can still have RGB but normalized to 1. RGB is more about having 3 channel than about what scale they are on.

File: mie2c/e2c.py (hunk 2 above)
Comment (blandry) [reply mie2c/e2c.py:375]: I think similar to last comment, I'm assuming that your dataset already comes in as 0-1 float32's. That's what it is for the Benchmark notebook. Is it different for the cartpole?

Comment (acauligi) [reply mie2c/e2c.py:375]: Cartpole is currently not being preprocessed, but I can make the data processing consistent once this gets approved. However, why would _x_dec_ require a _clamp_ here? The output of the decoder is already normalized 0-1.

Comment (blandry) [reply mie2c/e2c.py]: you are correct (fixed)
